plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -405,19 +405,32 @@ public function capture_authorized_payment( $order ) {
 	private function get_authorization_id_for_capture( $order ) {
 		$paypal_order_id  = $order->get_meta( '_paypal_order_id', true );
 		$authorization_id = $order->get_meta( '_paypal_authorization_id', true );
+		$capture_id       = $order->get_meta( '_paypal_capture_id', true );
 
-		if ( ! $paypal_order_id ) {
+		// If the PayPal order ID is not found or the capture ID is already set, return null.
+		if ( ! $paypal_order_id || ! empty( $capture_id ) ) {
 			return null;
 		}
 
-		// If the authorization ID is not found, try to retrieve it from the PayPal order details as a fallback for backwards compatibility.
-		if ( ! $authorization_id ) {
+		// If '_paypal_authorization_checked' is set to 'yes' and the authorization ID is empty, it means we already checked and found no authorization data.
+		// Return null to avoid repeated API calls.
+		if ( 'yes' === $order->get_meta( '_paypal_authorization_checked', true ) && empty( $authorization_id ) ) {
+			return null;
```

Comment:
Not blocking: do we want a debug entry for this case? I am not sure if we do, but surfacing regardless.